src/profiler/profiler.c

```diff
@@ -472,11 +472,13 @@ void xdebug_profiler_function_end(function_stack_entry *fse)
 		fse->profile.time -= call_entry->time_taken;
 		fse->profile.memory -= call_entry->mem_used;
 	}
-	xdebug_str_add_fmt(
-		&file_buffer, "%d %lu %lu\n",
-		fse->profiler.lineno,
-		(unsigned long) (fse->profile.time * 1000000), fse->profile.memory >= 0 ? fse->profile.memory : 0
-	);
+
+	xdebug_str_add_uint64(&file_buffer, fse->profiler.lineno);
+	xdebug_str_addc(&file_buffer, ' ');
+	xdebug_str_add_uint64(&file_buffer, (unsigned long) (fse->profile.time * 1000000));
+	xdebug_str_addc(&file_buffer, ' ');
+	xdebug_str_add_uint64(&file_buffer, fse->profile.memory >= 0 ? fse->profile.memory : 0);
+	xdebug_str_addc(&file_buffer, '\n');
```

Comment:
It does make it harder to read IMO. I think I would prefer a comment above these lines (as well as in the replacement section below), that says: `/* Adds %d %lu %lu, with lineno, time, and memory */`

Reply:
solved - declared twice, we may even put in `static inline` function